trieste/acquisition/_optimizer.py

```diff
@@ -62,11 +63,21 @@ def _box(space: Box, target_func: TensorMapping) -> QueryPoints:
     initial_point = optimize(trial_search_space, target_func)
 
     bijector = tfp.bijectors.Sigmoid(low=space.lower, high=space.upper)
-    variable = tf.Variable(bijector.inverse(initial_point))
 
-    def _objective() -> tf.Tensor:
-        return -target_func(bijector.forward(variable))
+    def _objective(x: tf.Tensor) -> tf.Tensor:
+        return tf.squeeze(-target_func(bijector.forward(x)), axis=-1)
 
-    gpflow.optimizers.Scipy().minimize(_objective, (variable,))
+    lbfgs_result = tfp.optimizer.lbfgs_minimize(
+        lambda x: tfp.math.value_and_gradient(_objective, x),
+        # we copy the tensor to work around a bug in tensorflow_probability,
+        #   see https://github.com/tensorflow/probability/issues/1182
+        tf.identity(bijector.inverse(initial_point)),
```

Comment:
note that while we're now explicitly working around a bug, the bug is not new in this PR (the bug is in tfp bijectors). It just wasn't breaking any tests before